---
title: C6312
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6312
helpviewer_keywords:
- C6312
ms.assetid: 1fc8b9a1-e6ba-4799-84c3-31f289576cca
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 5a67173e7621204122a26c2b0ebe632eb01000d0
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 04/23/2019
ms.locfileid: "62781666"
---
# <a name="c6312"></a>C6312
警告 C6312:可能是无限循环： 常量 exception_continue_execution 试一试的异常筛选器表达式中使用-除外

 此警告指示常量使用`EXCEPTION_CONTINUE_EXECUTION`（或另一个常量，它的计算结果为-1） 的结构化的异常处理程序的筛选器表达式中。 使用的常量值`EXCEPTION_CONTINUE_EXECUTION`可能会导致无限循环。 例如，如果硬件时引发了异常，将重启导致异常的说明。 如果仍错误导致异常的地址，另一个异常会发生，并在相同的方式进行处理。 这将导致无限循环。

 显式调用`RaiseException`不能直接将导致无限循环，但它将继续在受保护块的代码的执行。 这可能是意料之外，并可能导致无限循环，如果`RaiseException`用于避免取消引用了无效的指针。

 通常情况下，`EXCEPTION_CONTINUE_EXECUTION`仅在筛选器表达式中，有机会修复导致异常的指针，或者基础内存时调用的函数应返回。

## <a name="example"></a>示例
 下面的代码生成此警告：

```
#include <excpt.h>
#include <stdio.h>
#include <windows.h>

void f (char *ptr)
{
  __try
  {
    // exception occurs if the caller passes null ptr
    // code...
   *ptr = '\0';
  }
  __except (EXCEPTION_CONTINUE_EXECUTION)
    // When EXCEPTION_CONTINUE_EXECUTION is used, the handler
    //  block of the structured exception handler is not executed.
  {
    puts("This block is never executed");
  }
}
```

 若要更正此警告，请使用以下代码：

```
#include <excpt.h>
#include <stdio.h>
#include <windows.h>

void f (char *ptr)
{
  __try
  {
    // exception occurs if the caller passes null ptr
    // code...
    *ptr = '\0';
  }
  __except (GetExceptionCode()== EXCEPTION_ACCESS_VIOLATION ?
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)
  {
    puts("Error Occurred");
  }
}
```